### DISCUSSION TITLE
Whats the minimum needed to install to start with soruce generators?

### DISCUSSION THREAD

**Discussion by zbesevic** [dotnet/roslyn#50103] [Q&A]
Assume that I need to have only the compiler which has the phase for source generations and the nugets? So the projects target framework and lang version don't need to change on the assembly level? Is the compiler shipped with .NET 5 SDK?

**Comment by PathogenDavid** [comment]
You can develop and consume source generators with nothing but the .NET SDK.

**Comment by huoyaoyuan** [comment]
Source generator is a **compiler** feature, not **language** feature. It's **released** since VS16.8/.NET 5 SDK/Microsoft.CodeAnalysis 3.8.

To provide a source generator, you need to install the nuget package. To consume a source generator, you should use `global.json` or something to limit .NET SDK or VS version.

It's totally OK to use latest SDK to target older frameworks.
